zanata-war/src/main/java/org/zanata/security/SmartEntitySecurityListener.java

```diff
@@ -2,58 +2,57 @@
 
 import java.util.Arrays;
 import java.util.List;
+
 import javax.persistence.PostLoad;
 import javax.persistence.PrePersist;
 import javax.persistence.PreRemove;
 import javax.persistence.PreUpdate;
 
-import org.jboss.seam.security.EntityAction;
-import org.jboss.seam.security.EntitySecurityListener;
-import org.jboss.seam.security.Identity;
+import org.jboss.seam.contexts.Contexts;
+import org.jboss.seam.persistence.PersistenceProvider;
 import org.zanata.annotation.EntityRestrict;
 
 /**
- * Overrides EntitySecurityListener to avoid calling
- * EntityPermissionChecker.instance() when security is disabled, which leads to
+ * Disable permission checking when security is disabled, which solves the
  * problems if there is no active Seam application context (in tests).
  *
  * @author Sean Flanigan <a
  *         href="mailto:sflaniga@redhat.com">sflaniga@redhat.com</a>
  *
  */
-public class SmartEntitySecurityListener extends EntitySecurityListener {
+public class SmartEntitySecurityListener {
```

Comment:
Same way. Pre and Post are JPA/hibernate annotation. It just delegate the check to ZanataIdentity as before. Except this line we need to revisit:
```
PersistenceProvider provider = PersistenceProvider.instance();
Class beanClass = provider.getBeanClass(entity);
```
I think we are safe to just use entity.getClass(). the code snippet above just cater for entity being an EJB. And in our case it's always just a plain hibernate entity.